src/sentry/integrations/discord/message_builder/issues.py

```diff
@@ -0,0 +1,149 @@
+        obj: Group | GroupEvent = self.event if self.event is not None else self.group
+        rule_id = None
+        if self.rules:
+            rule_id = self.rules[0].id
+
+        embeds = [
+            DiscordMessageEmbed(
+                title=build_attachment_title(obj),
+                description=build_attachment_text(self.group, self.event) or None,
+                url=get_title_link(
+                    self.group,
+                    self.event,
+                    self.link_to_event,
+                    self.issue_details,
+                    self.notification,
+                    ExternalProviders.DISCORD,
+                    rule_id,
+                ),
+                color=LEVEL_TO_COLOR["info"],
+                # We can't embed urls in Discord embed footers.
+                footer=DiscordMessageEmbedFooter(
+                    build_footer(self.group, project, self.rules, "{text}")
+                ),
+                fields=build_tag_fields(event_for_tags, self.tags),
+                timestamp=timestamp,
+            )
+        ]
+
+        components = build_components(self.group, project)
+
+        return self._build(embeds=embeds, components=components)
+
+
+def build_tag_fields(
+    event_for_tags: GroupEvent | None, tags: set[str] | None = None
+) -> list[DiscordMessageEmbedField]:
+    fields: list[DiscordMessageEmbedField] = []
+    if tags:
+        event_tags = event_for_tags.tags if event_for_tags else []
+        for key, value in event_tags:
+            std_key = tagstore.get_standardized_key(key)  # type: ignore
+            if std_key not in tags:
+                continue
+
+            labeled_value = tagstore.get_tag_value_label(key, value)  # type: ignore
+            fields.append(
+                DiscordMessageEmbedField(
+                    std_key,
+                    labeled_value,
+                    inline=True,
+                )
+            )
+    return fields
+
+
+def build_components(
+    group: Group,
+    project: Project,
+) -> list[DiscordMessageComponent]:
+
+    archive_button = DiscordButton(
+        custom_id=f"{CustomIds.ARCHIVE}:{group.id}",
+        label="Archive",
+    )
+
+    resolve_button = DiscordButton(
+        custom_id=f"{CustomIds.RESOLVE_DIALOG}:{group.id}", label="Resolve..."
+    )
+
+    assign_button = DiscordButton(
+        custom_id=f"{CustomIds.ASSIGN_DIALOG}:{group.id}", label="Assign..."
+    )
+
+    status = group.get_status()
+
+    if not has_releases(project):
+        resolve_button = DiscordButton(
+            custom_id=f"{CustomIds.RESOLVE}:{group.id}",
+            label="Resolve",
+        )
+
+    if status == GroupStatus.RESOLVED:
+        resolve_button = DiscordButton(
+            custom_id=f"{CustomIds.UNRESOLVE}:{group.id}",
+            label="Unresolve",
+        )
+
+    if status == GroupStatus.IGNORED:
+        archive_button = DiscordButton(
+            custom_id=f"{CustomIds.MARK_ONGOING}:{group.id}",
```

Comment:
We may need to check the escalating issues/issue states flag here - If the feature is disabled, this should be "Mark Unresolved".